addons/mogdb/dataprotection/gs-basebackup-backup.sh

```diff
@@ -0,0 +1,40 @@
+#set -e
+#set -o pipefail
+export PATH="$PATH:$DP_DATASAFED_BIN_PATH"
+export DATASAFED_BACKEND_BASE_PATH="$DP_BACKUP_BASE_PATH"
+export PGPASSWORD=${DP_DB_PASSWORD}
+
+trap handle_exit EXIT
+
+START_TIME=`get_current_time`
+BACKUP_TMPDIR="/tmp"
+
+if [ ! -f /usr/bin/expect ]; then
+  OLD_LD_LIBRARY_PATH="${LD_LIBRARY_PATH}"
+  unset LD_LIBRARY_PATH
+
+  apt-get update && DEBIAN_FRONTEND=noninteractive apt-get install -y expect
+
+  export LD_LIBRARY_PATH="${OLD_LD_LIBRARY_PATH}"
+fi
+
+/usr/bin/expect <<-EOF
+set time 30
+spawn gs_basebackup -Ft -Pv -c fast -Xf -D ${BACKUP_TMPDIR} -h ${DP_DB_HOST} -U ${DP_DB_USER} -z -W
```

Comment:
it is better to use the compression method in datasafed, the backup speed is very slow. you can reference the `postgresql` addon. and better to stream upload to backupRepo.

Reply:
mogdb do not support stream backup and not support pipe password

Reply:
oh, if mogdb support PGPASSWORD enviroment, you can use datasafed to stream upload the base.tar by pipe.

Reply:
1. Also not support PGPASSWORD enviroment.
2. gs_basebackup command has a limitation that a valid directory must be provided('-' is not valid).
3. Do you mean to run follow command:
	1) spawn gs_basebackup -Ft -Pv -c fast -Xf -D ${BACKUP_TMPDIR} -h ${DP_DB_HOST} -U ${DP_DB_USER} -W
	2) datasafed push -z zstd-fastest - "/${DP_BACKUP_NAME}.tar.zst" < "$BACKUP_TAR_FILE"